File: spring-boot-portail-client-banque/portailClient/src/main/java/com/example/portailClient/model/ObjectResponseData.java
package com.example.portailClient.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class ObjectResponseData {
    private Long im_familles_id;
    private String libelle;
    private Long im_sous_familles_id;
    private String sf_libelle;
    private Long sf_im_familles_id;
    private Boolean vendable;
    private Boolean classe_ngp_libre;
    private String abr;
    private ImArticle imArticle;
}

File: portail-client-banque/src/app/model/ObjectResponseData.ts
import { ImArticle } from "./ImArticle";

export class ObjectResponseData{
    constructor(
        public im_familles_id:number,
        public libelle:string,
        public im_sous_familles_id:number,
        public sf_libelle:string,
        public sf_im_familles_id:number,
        public vendable:boolean,
        public classe_ngp_libre:boolean,
        public abr:string,
        public imArticle:ImArticle
    ){}
}